app/components/user-page/course-progress-list-item/index.ts

```diff
@@ -0,0 +1,69 @@
+import Component from '@glimmer/component';
+import arrayToSentence from 'array-to-sentence';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import uniqReducer from 'codecrafters-frontend/utils/uniq-reducer';
+import fieldComparator from 'codecrafters-frontend/utils/field-comparator';
+import type RouterService from '@ember/routing/router-service';
+import type CourseParticipationModel from 'codecrafters-frontend/models/course-participation';
+
+interface Signature {
+  Element: HTMLDivElement;
+
+  Args: {
+    courseParticipations: CourseParticipationModel[];
+  };
+}
+
+export default class CourseProgressListItem extends Component<Signature> {
+  @service declare router: RouterService;
+
+  get completedAt() {
+    return this.completedCourseParticipations[0]?.completedAt || null;
+  }
+
+  get completedCourseParticipations() {
+    return this.args.courseParticipations.filter((item) => item.isCompleted);
+  }
+
+  get completedStagesCount() {
+    return Math.max(...this.args.courseParticipations.map((item) => item.completedStageSlugs).map((stage_list) => stage_list.length));
+  }
+
+  get course() {
+    return this.args.courseParticipations[0]?.course;
```

Comment:
```suggestion
    return this.args.courseParticipations[0]!.course;
```

We know that this won't be instantiated without at least one course participation in the group